common/src/main/java/org/apache/celeborn/common/write/LocationPushFailedBatches.java

```diff
@@ -0,0 +1,94 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.celeborn.common.write;
+
+import java.io.Serializable;
+import java.util.*;
+import java.util.concurrent.ConcurrentHashMap;
+
+import org.apache.commons.lang3.StringUtils;
+
+import org.apache.celeborn.common.util.JavaUtils;
+import org.apache.celeborn.common.util.Utils;
+
+public class LocationPushFailedBatches implements Serializable {
+
+  // map-attempt-id -> failed batch ids
+  private final Map<String, Set<Integer>> failedBatches = JavaUtils.newConcurrentHashMap();
```

Comment:
Is it better to use `Map<Tuple2<String, String>, Set<Integer>`?

Reply:
Java is a strongly typed language, and generic tuples don't have the benefit of strong typing. This LocationPushFailedBatches is a Java class, which I think we should involve a Scala tuple unless necessary.